Document Package Source Mapping error NU1110 (#3169)
NuGet/docs.microsoft.com-nuget: docs/reference/errors-and-warnings/NU1110.md

---
title: NuGet Error NU1110
description: NU1110 error code
author: donnie-msft
ms.author: eagoodso
ms.date: 10/27/2023
ms.topic: reference
f1_keywords: 
  - "NU1110"
---

# NuGet Error NU1110

<pre>NU1110: The package `B` is available in the Global packages folder, but the source it came from `https://api.nuget.org/v3/index.json` is not one of the configured sources.
</pre>

### Issue

A previous restore retrieved package `B` from a package source which your current solution does not have configured.

> While performing a package installation of package `A`, a dependent package `B` was needed transitively to complete the action. Restore found `B` in your machine's [Global packages folder](https://aka.ms/troubleshoot_nuget_cache), and the accompanying `nupkg.metadata` indicates a `Source` value which it cannot find in your solution.

Reviewer: ```suggestion
While performing a package installation of package `A`, a dependent package `B` was needed transitively to complete the action. Restore found `B` in your [Global packages folder](../../consume-packages/managing-the-global-packages-and-cache-folders), and the accompanying `nupkg.metadata` indicates a `Source` value which it cannot find in your solution.
```

The global packages folder is user specific. I think we don't need extra specificity. 

Also the guidance for referencing internal docs is to reference the location internally.

Author: Done 👍

> When [package source mapping](https://aka.ms/nuget-package-source-mapping) is enabled, NuGet attempts to map transitive dependencies automatically by analyzing the source in the Global packages folder. 

Reviewer: same here.

Author: Done 👍

> ### Solution

Reviewer: Not something we'd recommend frequently, but wouldn't a 3rd option be to add that source to your config? 
There's a chance that package `B` is not available in any of the configured sources.

Author: Made an attempt at writing option 3.

---
title: NuGet Error NU1110
description: NU1110 error code
author: donnie-msft
ms.author: eagoodso
ms.date: 10/27/2023
ms.topic: reference
f1_keywords: 
  - "NU1110"
---

# NuGet Error NU1110

<pre>NU1110: The package `B` is available in the Global packages folder, but the source it came from `https://api.nuget.org/v3/index.json` is not one of the configured sources.
</pre>

### Issue

A previous restore retrieved package `B` from a package source which your current solution does not have configured.

> While performing a package installation of package `A`, a dependent package `B` was needed transitively to complete the action. Restore found `B` in your [Global packages folder](../../consume-packages/managing-the-global-packages-and-cache-folders), and the accompanying `nupkg.metadata` indicates a `Source` value which it cannot find in your solution.

Reviewer: I looked at some of the other links in the repo and it seems like most of them are using .md

I think that's why Errors-And-Warnings validation passes, while this one doesn't.

> When [package source mapping](../../../consume-packages/package-source-mapping.md) is enabled, NuGet attempts to map transitive dependencies automatically by analyzing the source in the Global packages folder.

Reviewer: I think you have an an extra `..`

```suggestion
When [package source mapping](../../consume-packages/package-source-mapping.md) is enabled, NuGet attempts to map transitive dependencies automatically by analyzing the source in the Global packages folder.
```

Author: The host document is `docs/reference/errors-and-warnings/NU1110.md`
The documents I'm trying to link should be children of `docs`. I think I actually should want 3 levels? I'm very confused now. Either way, validation is failing, but with only 2 levels, the preview's links actually work for me.

### Solution

#### Option 1. 
Create a package source mapping for package `B`.

#### Option 2.

> Clear package `B` from your Global packages folder so that restore can download from one of your configured packages sources. For more information, see [Managing the global packages, cache, and temp folders](../../../consume-packages/managing-the-global-packages-and-cache-folders.md).

Reviewer: ```suggestion
Clear package `B` from your Global packages folder so that restore can download from one of your configured packages sources. For more information, see [Managing the global packages, cache, and temp folders](../../consume-packages/managing-the-global-packages-and-cache-folders.md).
```

Author: First time, I had the right path. I then added `.md` but accidentally left my test of 3 levels up. Trying again..

If your other solutions have different package sources and package source mapping configurations, this option may not be feasible.

#### Option 3.

> Consider enabling a package source which you know has package `B` available.

Reviewer: adding? enabling? I'm torn. 

Feels like adding covers more scenarios, but I'll let you make a call.

Author: I think enabling implies both and uses the verbiage that's more specific to package sources.
